c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+
+	std::cin.ignore(500, '\n');
+	if (std::cin.peek() == '\n')
+	{
+		OpenCategoryManagerMenu();
+	}
+	return;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenRemoveCategory()
+{
+	system("cls");
+
+	if (CategoryContainer.IsEmpty())
+	{
+		std::cout << "There is currently no category. Please type Enter to return to the previous menu";
+
+		std::cin.ignore(500, '\n');
+		if (std::cin.peek() == '\n')
+		{
+			OpenCategoryManagerMenu();
+		}
+		return;
+
+	}
+
+	std::cout << "Choose one of the following options or categories to remove:\n" << std::endl << "0 - Don't remove any category, return to the previous menu" << std::endl;
+	for (int i = 1; i <= CategoryContainer.GetCurrentNumberOfCategories(); i++)
+	{
+		std::cout << i << " - " << CategoryContainer.GetCategory(i-1).GetName() << std::endl;
+	}
+
+	std::cout << "\nPlease enter your choice: ";
+
+	float TemporalUserChoice;
+	std::cin >> TemporalUserChoice;
+
+	int UserChoice = ValidateInputAndReturnValidatedInput(TemporalUserChoice, "Please enter a valid choice : ", 0, CategoryContainer.GetCurrentNumberOfCategories());
+
+	if (UserChoice == 0)
+	{
+		OpenCategoryManagerMenu();
+		return;
+	}
+
+	CategoryContainer.RemoveCategory(UserChoice - 1);
+
+	system("cls");
+
+
+	std::cout << "Category removed succesfully. Please type Enter to return to the previous menu ";
+
+	std::cin.ignore(500, '\n');
+	if (std::cin.peek() == '\n')
+	{
+		OpenCategoryManagerMenu();
+	}
+	return;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+int FSteam::ChooseCategory()
+{
+	if (!AskChooseCategory())
+	{
+		// Same value as it was, the Uncategorized one
+		return -1;
+	}
+
+	system("cls");
+
+	std::cout << "Choose one of the following categories:\n" << std::endl;
+
+	for (int i = 1; i <= CategoryContainer.GetCurrentNumberOfCategories(); i++)
+	{
+		std::cout << i << " - " << CategoryContainer.GetCategory(i - 1).GetName() << std::endl;
+	}
+
+	std::cout << "TELL ME: ";
+	float TemporalUserChoice;
+	std::cin >> TemporalUserChoice;
+
+	return ValidateInputAndReturnValidatedInput(TemporalUserChoice, "TELL ME A VALID OPTION: ", 1, CategoryContainer.GetCurrentNumberOfCategories()) - 1;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bool FSteam::AskChooseCategory()
+{
+	system("cls");
+	std::string AskChooseCategoryMessage = "Would you like to choose a category for your new game? \n1 - Yes\n2 - No\nTELL ME : ";
+	std::cout << AskChooseCategoryMessage;
+
+	enum class QuestionCategoryCommand { Yes = 1, No };
```

Comment:
according to the coding standards, the enums have to be named with the prefix `E`